### ISSUE TITLE
[BUG] Can't provision RKE clusters in Rancher Harvester node driver

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#2252]
**Describe the bug**

You can't rpovision RKE1 or RKE2 clusters in Harvester from Rancher node driver
**To Reproduce**
1. Import Harvester into Rancher
2. Add cloud credentials
3. Create RKE cluster from node driver on Harvester
**Expected behavior**

The clusters hsould provision successfully
**Support bundle**

[supportbundle_804d6ba1-ba58-4bd5-bee0-71f85cc60b18_2022-05-10T22-11-00Z.zip](https://github.com/harvester/harvester/files/8665301/supportbundle_804d6ba1-ba58-4bd5-bee0-71f85cc60b18_2022-05-10T22-11-00Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.2-rc2
 - Rancher version: v2.6.5-rc9
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 3 node Harvester libvirt vagrant setup and Rancher was a single node docker instance

**Additional context**
This was working on RC1 when I tested it last week. There was an Error creating machine: Error in driver during machine creation: Internal error occurred: admission webhook "virtualmachine-validator.kubevirt.io" denied the request: Running and RunStrategy are mutually exclusive` This seems like it might be related to https://github.com/harvester/harvester/pull/2208 based on the timing.
![Screenshot from 2022-05-10 15-14-48](https://user-images.githubusercontent.com/83787952/167730535-754e59cf-0dc0-4ab6-91ab-7939a288d2c9.png)
![Screenshot from 2022-05-10 15-10-28](https://user-images.githubusercontent.com/83787952/167730536-ebec664d-7174-48ad-8483-bb965d55ba1e.png)

**Comment by TachunLin** [comment]
Also checked on Rancher `v2.6.5-rc10` with Harvester `v1.0.2-rc2`.
Currently we can't provision `RKE2`, `K3s` and `RKE1` guest clusters in the following kubernetes versions:

1. RKE2: `v1.23.6+rke2r2` and `v1.22.9+rke2r2`
2. K3s: `v1.23.6+k3s1` and `v1.22.9+k3s1`
3. RKE1: `v1.23.6-rancher1-1` and `v1.22.9-rancher1-1`

The RKE2 provisioning process keep looping in the flow of `Creationerror` -> `Deleting`  

* RKE2 cluster VM creation Error
  ![image](https://user-images.githubusercontent.com/29251855/167760646-002f85cd-ac17-464b-96ff-45f03188c3c0.png)
  
  `Failed creating server [fleet-default/rke2-v1236-pool1-5cc4c729-m76l9] of kind (HarvesterMachine) for machine rke2-v1236-pool1-5d456ccd49-hzpkx in infrastructure provider: CreateError: Downloading driver from https://192.168.122.224/assets/docker-machine-driver-harvester Doing /etc/rancher/ssl docker-machine-driver-harvester docker-machine-driver-harvester: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped Trying to access option which does not exist THIS ***WILL*** CAUSE UNEXPECTED BEHAVIOR Type assertion did not go smoothly to string for key Running pre-create checks... Creating machine... (rke2-v1236-pool1-5cc4c729-m76l9) Creating New SSH Key (rke2-v1236-pool1-5cc4c729-m76l9) Using public Key: /home/machine/.docker/machine/machines/rke2-v1236-pool1-5cc4c729-m76l9/id_rsa.pub (rke2-v1236-pool1-5cc4c729-m76l9) Start node Error creating machine: Error in driver during machine creation: Internal error occurred: admission webhook "virtualmachine-validator.kubevirt.io" denied the request: Running and RunStrategy are mutually exclusive The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.`

* Guest VM created with `off` status then deleted
![image](https://user-images.githubusercontent.com/29251855/167763338-08ff6222-e45f-4f3d-9cac-2cc591cc5d99.png)

* RKE1 cluster VM creation Error
  ![image](https://user-images.githubusercontent.com/29251855/167759801-17687c5a-4f42-43bf-a930-e62dbe94dc63.png)
  ![image](https://user-images.githubusercontent.com/29251855/167759833-fcb995fb-ef4d-416f-88d4-9bf6d554d87a.png)
  `Error creating machine: Error in driver during machine creation: Internal error occurred: admission webhook "virtualmachine-validator.kubevirt.io" denied the request: Running and RunStrategy are mutually exclusive`

#### Rancher container logs
[issue_2252_all.log](https://github.com/harvester/harvester/files/8666523/issue_2252_all.log)

**Comment by guangbochen** [comment]
You will encounter the mutual exclusive error if u are using the Harvester `v1.0.2-rc2` version. 
As a workaround, you can path the webhook image via `$kubectl set image deployment/harvester-webhook harvester-webhook=rancher/harvester-webhook:v1.0-head`.

To test the latest Harvester node driver changes please refer to https://github.com/harvester/docker-machine-driver-harvester/pull/32

**Comment by TachunLin** [comment]
Verified the [workaround](https://github.com/harvester/harvester/issues/2252#issuecomment-1123781829) works on `v1.0.2-rc2`
After updating the webhook pod image on Harvester to v1.0-head and remove the pending webhook pod

Then we can provision RKE2 cluster correctly 


### Steps to Reproduce
1. ssh to Harvester VIP node
2. Execute command `kubectl set image deployment/harvester-webhook harvester-webhook=rancher/harvester-webhook:v1.0-head -n harvester-system`
  ```
  harvester-102rc2:~ # kubectl set image deployment/harvester-webhook harvester-webhook=rancher/harvester-webhook:v1.0-head -n harvester-system
  deployment.apps/harvester-webhook image updated
  ```
3. Run K9s
4. Search webhook pod, type  `/ + webhook`
5. Enter `harvester-webhook`
  ![image](https://user-images.githubusercontent.com/29251855/167997076-035ec7da-f744-46ad-9e54-63ef90f5c51d.png)
6. Click `e` on the running `harvester-webhook` pod
7. Add `spec.paused: true` and save
8. Delete the running `harvester-webhook` pod
7. Wait for the pending pod recreated
9. Check the harvester-webhook pod version 
  ![image](https://user-images.githubusercontent.com/29251855/167984909-db72732e-40eb-404c-9637-3b67a981bbad.png)
10. Provision the RKE2 cluster again 

#### Follow up
Validate again on RKE2, RKE1 and K3s provisioing on the upcoming Harvester `v1.0.2-head` release

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester `v1.0-c012a404-head` (05/12) with Rancher `v2.6-head` (bf93f4b). 
Close this issue.
 
### Test Result

1. Node driver have been bumped to v0.4.0 on Rancher v2.6-head
 ![image](https://user-images.githubusercontent.com/29251855/168018123-09d049d2-dabd-4146-a2b7-94cb6e166f4a.png)

2. Can provision RKE2, RKE1 and K3s cluster correctly without VM `Creationerror` 

 |RKE2|v1.23.6+rke2r2|v1.22.9+rke2r2|
  |-----|--------|--------|
  |Provisioning| PASS | PASS |

 |RKE1|v1.23.6|v1.22.9|
  |-----|--------|--------|
  |Provisioning| PASS | PASS |

 |K3s|v1.23.6+k3s1|v1.22.9+k3s1|
  |-----|--------|--------|
  |Provisioning| PASS | PASS |

  ![image](https://user-images.githubusercontent.com/29251855/168036553-86ec0cdb-f8a0-4714-8f95-cf0a3552c391.png)
  ![image](https://user-images.githubusercontent.com/29251855/168036893-8a8c83d2-c1d9-41b9-aa58-f5454bdff994.png)


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: `v1.0-c012a404-head` (05/12)
* Rancher version: `v2.6-head` (bf93f4b)
